### ISSUE TITLE
How to avoid re-rendering of whole form?

### ISSUE THREAD

**Issue by niphidrema** [davidkpiano/react-redux-form#656]
I have little bit bigger form with lot of custom components.
I need to save change to model on every change not only on focus.
However on every call to onChange() , whole form is rerendered instead of only changed component and this is causing big lags.
Should I cache current value in all components and create shouldComponentUpdate() ? 
Or is there any way this can be solved directly in your library (I would use it , if is).

EDIT:
Ok I found out your library is impossible to use with anything bigger liek few imputs because it is simply trying to re-render everything on every dispatch or onChange even pure components..

Here is profiler:
![image](https://cloud.githubusercontent.com/assets/25751004/22903844/70be8914-f23a-11e6-83da-faaa44db9e1c.png)

**Comment by davidkpiano** [comment]
Can I see your code? Are you `connect()`-ing the entire form?

**Comment by davidkpiano** [comment]
Also, you can simply `updateOn="blur"` which will immediately give you huge performance benefits.

**Comment by davidkpiano** [comment]
Some of the performance concerns have been addressed in this commit: bf53863 which include many improvements:

- actions.promise will now treat rejected promise errors as async errors
- model actions are external by default - only when they are called within a Control event handler are they considered internal.
- no validation will occur on Controls if there are no validators/errors props.
- no validation will occur on Form if there are no validators/errors props unless the form is made invalid (async) - then the validity will become valid again on next validation.
- all validation originating from Form is merged by default with field validation.
- isValid() correctly handles async simple validation (strings, boolean, etc.)
- no ‘validate’ intent will be pushed if change is not external (originated from Control)

In short, this means that fewer actions are dispatched now, which improves overall performance. More updates to come!

**Comment by davidkpiano** [comment]
@niphidrema The issue could be that you are `connect()`-ing your entire form. This will, of course, cause the entire thing to update, and is not necessary. Without seeing your code, I cannot know for sure. I'll reopen this if you fixed that and are still having problems.
